node/malus/src/variants/suggest_garbage_candidate.rs

```diff
@@ -88,131 +92,166 @@ where
+								.await
+							{
+								Ok(Some((validation_data, validation_code))) => {
+									sender
+										.send((validation_data, validation_code, n_validators))
+										.expect("channel is still open");
+								},
+								_ => {
+									panic!("Unable to fetch validation data");
+								},
+							}
+						}),
+					);
 
-				let message = FromOrchestra::Communication {
-					msg: CandidateBackingMessage::Second(relay_parent, malicious_candidate, pov),
-				};
+					let (validation_data, validation_code, n_validators) = receiver.recv().unwrap();
+
+					let validation_data_hash = validation_data.hash();
+					let validation_code_hash = validation_code.hash();
+					let validation_data_relay_parent_number = validation_data.relay_parent_number;
+
+					gum::trace!(
+						target: MALUS,
+						candidate_hash = ?candidate.hash(),
+						?relay_parent,
+						?n_validators,
+						?validation_data_hash,
+						?validation_code_hash,
+						?validation_data_relay_parent_number,
+						"Fetched validation data."
+					);
 
-				Some(message)
+					let malicious_available_data =
+						AvailableData { pov: Arc::new(pov.clone()), validation_data };
+
+					let pov_hash = pov.hash();
+					let erasure_root = {
+						let chunks = erasure::obtain_chunks_v1(
+							n_validators as usize,
+							&malicious_available_data,
+						)
+						.unwrap();
+
+						let branches = erasure::branches(chunks.as_ref());
+						branches.root()
+					};
+
+					let (collator_id, collator_signature) = {
+						use polkadot_primitives::v2::CollatorPair;
+						use sp_core::crypto::Pair;
+
+						let collator_pair = CollatorPair::generate().0;
+						let signature_payload = polkadot_primitives::v2::collator_signature_payload(
+							&relay_parent,
+							&candidate.descriptor().para_id,
+							&validation_data_hash,
+							&pov_hash,
+							&validation_code_hash,
+						);
+
+						(collator_pair.public(), collator_pair.sign(&signature_payload))
+					};
+
+					let malicious_commitments = create_fake_candidate_commitments(
+						&malicious_available_data.validation_data,
+					);
+
+					let malicious_candidate = CandidateReceipt {
+						descriptor: CandidateDescriptor {
+							para_id: candidate.descriptor().para_id,
+							relay_parent,
+							collator: collator_id,
+							persisted_validation_data_hash: validation_data_hash,
+							pov_hash,
+							erasure_root,
+							signature: collator_signature,
+							para_head: malicious_commitments.head_data.hash(),
+							validation_code_hash,
+						},
+						commitments_hash: malicious_commitments.hash(),
+					};
+					let malicious_candidate_hash = malicious_candidate.hash();
+
+					gum::debug!(
+						target: MALUS,
+						candidate_hash = ?candidate.hash(),
+						?malicious_candidate_hash,
+						"Created malicious candidate"
+					);
+
+					// Map malicious candidate to the original one. We need this mapping to send back the correct seconded statement
```

Comment:
I think this never actually happens (not implemented). I suggest to remove it, as well as the impl for the outgoing message, but keep the comments about what happens when we send back an unexpected collation seconded message.

Reply:
So does this mean we can also remove `Inner` ?

Reply:
if it's empty, yeah we should just remove that.